### PR TITLE
Update omnibus toolchain for AIX, enable toolchain shell on AIX

### PR DESCRIPTION
**Pull request by scotthain**
Replaces #136

### CODE REVIEW

**Comment by scotthain** [comment]
@chef-cookbooks/engineering-services @chef-cookbooks/omnibus-maintainers

**Comment by curiositycasualty** [comment]
First was a transient failure on Travis, reran travis test. :+1:

File: recipes/_omnibus_toolchain.rb [hunk 1]
```diff
@@ -21,6 +21,11 @@
 toolchain_name = node['omnibus']['toolchain_name']
 toolchain_version = node['omnibus']['toolchain_version']
 
+if File.exist?('/opt/omnibus-toolchain') || File.exist?('/opt/angry-omnibus-toolchain')
```
Comment (schisamo) [inline recipes/_omnibus_toolchain.rb]: Code flow wise we should drop this `installed` variable and just place everything after this in an `else`:

```ruby
if File.exist?('/opt/omnibus-toolchain') || File.exist?('/opt/angry-omnibus-toolchain')
  Chef::Log.warn('Assuming the existence of /opt/(angry-)omnibus-toolchain means that the package is already installed.')
else
    # ...
    remote_file package_path do
    source package_url
    action :create_if_missing
    not_if { installed }
  end
  
  package node['omnibus']['toolchain_name'] do
    source package_path
    options install_options
    not_if { installed }
  end
end
```

All that being said `remote_file` and `package` should be idempotent so placing _any_ extra guard logic in seems frivolous.

**Comment by scotthain** [comment]
@chef-cookbooks/omnibus-maintainers @chef-cookbooks/engineering-services ping...

File: recipes/_omnibus_toolchain.rb [hunk 2]
```diff
@@ -21,31 +21,41 @@
...
+    end
 
-  case node['kernel']['machine']
-  when 'i86pc'
-    package_url = "https://chef-releng.s3.amazonaws.com/omnibus/omnibus-toolchain/#{toolchain_name}-#{toolchain_version}-1.i86pc.solaris"
-  when 'sun4v', 'sun4u'
-    package_url = "https://chef-releng.s3.amazonaws.com/omnibus/omnibus-toolchain/#{toolchain_name}-#{toolchain_version}-1.sun4v.solaris"
+    install_options = '-a auto-install'
+  elsif aix?
+    package_url = "http://chef-releng.s3.amazonaws.com/omnibus/omnibus-toolchain/#{toolchain_name}-#{toolchain_version}-1.powerpc.bff"
```
Comment (schisamo) [inline recipes/_omnibus_toolchain.rb]: You should define a value for `install_options` globally in this recipe or in this branch of code.

File: recipes/_user.rb [hunk 3]
```diff
@@ -35,12 +35,18 @@
   ignore_failure true if windows?
 end
 
+# Ensure '/usr/local/bin/bash' is added to acceptable shells list
+execute 'chsec_login_shell' do
+  command "sudo chsec -f /etc/security/login.cfg -s usw -a 'shells=/bin/sh,/bin/bsh,/bin/csh,/bin/ksh,/bin/tsh,/bin/ksh93,/usr/bin/sh,/usr/bin/bsh,/usr/bin/csh,/usr/bin/ksh,/usr/bin/tsh,/usr/bin/ksh93,/usr/bin/rksh,/usr/bin/rksh93,/usr/sbin/uucp/uucico,/usr/sbin/sliplogin,/usr/sbin/snappd,/usr/bin/bash,#{node['omnibus']['build_user_shell']}'"
```
Comment (schisamo) [inline recipes/_user.rb]: Is the `sudo` required? Chef is almost always run with elevated privileges.

File: recipes/_omnibus_toolchain.rb (hunk 2 above)
Comment (scotthain) [inline recipes/_omnibus_toolchain.rb]: Not sure if I follow you on this one - it's only used for Solaris 10. If I understand you correctly, I could either add a `install_options = ''` or do something like:
```
  package node['omnibus']['toolchain_name'] do
    source package_path
    options '-a auto-install' if solaris?
  end
```

Comment (schisamo) [inline recipes/_omnibus_toolchain.rb]: You are assuming `install_options` is defined when this code is executed on every platform. On AIX `install_options` will not be defined so you either need to declare a default value (for all platforms) or define it in the `if/elsif` for each individual platform.

**Comment by schisamo** [comment]
@scotthain This looks good. Let's squash down to a single commit and :shipit:

**Comment by scotthain** [comment]
Thanks @schisamo - will merge pending travis happiness.

**Pull request merged**